iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalGrammar.java

```diff
@@ -130,6 +137,12 @@ private static void lexical(LexerlessGrammarBuilder b) {
     b.rule(PARAM_PREFIX).is(Punctuator.MINUS, Punctuator.MINUS);
     b.rule(PARAM_NAME).is(b.regexp("[a-z][-a-z]*+"));
     b.rule(PARAM_VALUE).is(b.regexp("[^\\s]+"));
+
+    b.rule(USER_STRING).is(b.regexp("(?:[a-z][-a-z0-9_]*|[0-9]+)"));
+    b.rule(USER_VARIABLE).is(b.regexp("\\$(?:[a-zA-Z_][a-zA-Z0-9_]*|\\{[^}]+\\})"));
```

Comment:
These regexes will not match `USER foo${var}` which is also valid. However, I guess it will only be solved with [SONARIAC-486](https://sonarsource.atlassian.net/browse/SONARIAC-486).

Reply:
Agree on this. I followed the grammar too strictly and it does not support such use case, but anyway we will need to review our way of handling arguments, so I prefer to drop it there.